### PR TITLE
Lint / hint about using Python wheels as sources

### LINKED ISSUE #1935: lint on most wheel-based python installs except `noarch: python` and `none-any.whl`
There are now several feedstocks that use pure python wheels with `noarch: python` recipes. This procedure is in principle safe, but an sdist is preferred. We should lint on any usage of wheels outside this one case. A second PR might also provide a hint that sdists are preferred.

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Added a ``news`` entry

Closes #1935

### CODE REVIEW

**Review by beckermr — APPROVED** [review]
We may want stronger language. Maybe "All packages should be built from source except in rare and exceptional cases."

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -693,6 +693,42 @@ def check_pins_build_and_requirements(top_level):
...
+            for line in f:
+                if match := pure_python_wheel_re.search(line):
+                    pure_python_wheel_urls.append(match.group(1))
+                elif match := wheel_re.search(line):
+                    compiled_wheel_urls.append(match.group(1))
+    if compiled_wheel_urls:
+        formatted_urls = ", ".join([f"`{url}`" for url in compiled_wheel_urls])
+        lints.append(
+            f"Detected compiled wheel(s) in source: {formatted_urls}. "
+            "This is discouraged. Please consider using a source distribution (sdist) instead."
```
Comment (jaimergp) [inline conda_smithy/lint_recipe.py:714]:
```suggestion
            "This is disallowed. All packages should be built from source except in "
            "rare and exceptional cases."
```

Comment (jaimergp) [reply conda_smithy/lint_recipe.py:714]: Something like this @beckermr?

**Comment by beckermr** [comment]
Yeah that's great.

**Comment by beckermr** [comment]
Thank you!

**Pull request merged**